Projects/Chess/Pieces/MoveOutcome.cs

```diff
@@ -0,0 +1,15 @@
+﻿namespace Chess;
+
+public class MoveOutcome
+{
+	public MoveOutcome()
+	{
+		IsSuccess = true;
+	}
+
+	public bool IsSuccess { get; set; }
+
+	public bool Attack { get; set; }
+
+	public Piece CapturedPiece { get; set; }
+}
```

Comment:
This one I told above. So I'd make it nullable where needed, removed `IsSuccess`, and the rest to be made into `record`